src/containers/shared/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,81 @@
+import classnames from 'classnames'
+import { useCallback, useEffect, useRef, useState } from 'react'
+import ArrowIcon from '../../images/down_arrow.svg'
+import './dropdown.scss'
+
+export interface DropdownProps {
+  title: string | JSX.Element
+  children: any
+  className?: string
+}
+
+// TODO: Add useId after upgrading to react@18 to populate id on .dropdown-menu and aria-controlled by on .dropdown-toggle
+/**
+ * A simple dropdown that has auto closing
+ *
+ * @param title The value in the toggle
+ * @param children The contents of the menu. DropdownItem is the preferred child component
+ * @param className
+ * @constructor
+ *
+ * @example
+ * <Dropdown title="Choose Something">
+ *   <DropdownItem handler={() => alert('hello')}>Option 1</DropdownItem>
+ *   <DropdownItem>Option 2</DropdownItem>
+ * </Dropdown>
+ */
+export const Dropdown = ({ title, children, className }: DropdownProps) => {
+  const [expanded, setExpanded] = useState(false)
+  const dropdownRef = useRef<HTMLDivElement>(null)
+
+  const globalClickListener = useCallback((nativeEvent) => {
+    // ignore click event happened inside the dropdown menu
+    if (dropdownRef.current && dropdownRef.current.contains(nativeEvent.target))
+      return
+    // else hide dropdown menu
+    setExpanded(false)
+    document.removeEventListener('click', globalClickListener)
+  }, [])
+
+  useEffect(
+    () => () => document.removeEventListener('click', globalClickListener),
```

Comment:
I might add a comment here and use an explicit return statement to make it less confusing for react newbies

Reply:
Done